src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -3228,7 +3228,10 @@ mongoc_collection_create_bulk_operation_with_opts (
                                       write_flags,
                                       wc);
 
-   mongoc_bulk_operation_set_let (bulk, &bulk_opts.let);
+   if (!bson_empty (&bulk_opts.let)) {
```

Comment:
`_mongoc_bulk_opts_parse` already initializes `bulk_opts.let`, so we need not use `bson_empty0`